Src/WitsmlExplorer.Api/Workers/Copy/CopyWbGeometrySectionsWorker.cs

```diff
@@ -0,0 +1,86 @@
+            }
+            if (componentsToCopy.Count() != job.Source.ComponentUids.Length)
+            {
+                string missingUids = string.Join(", ", componentsToCopy.Select((ts) => ts.Uid).Where((uid) => !job.Source.ComponentUids.Contains(uid)));
+                string reason = $"Could not retrieve all wbGeometrySections, missing uids: {missingUids}.";
+                Logger.LogError("{errorMessage} {reason} - {description}", errorMessage, reason, job.Description());
+                return (new WorkerResult(witsmlClient.GetServerHostname(), false, errorMessage, reason), null);
+            }
+            WitsmlWbGeometrys updatedWbGeometryQuery = WbGeometryQueries.CopyWbGeometrySections(targetWbGeometry, componentsToCopy);
+            QueryResult copyResult = await witsmlClient.UpdateInStoreAsync(updatedWbGeometryQuery);
+            string wbGeometrySectionsString = string.Join(", ", job.Source.ComponentUids);
+            if (!copyResult.IsSuccessful)
+            {
+                Logger.LogError("{errorMessage} - {job.Description()}", errorMessage, job.Description());
+                return (new WorkerResult(witsmlClient.GetServerHostname(), false, errorMessage, copyResult.Reason), null);
+            }
+
+            Logger.LogInformation("{JobType} - Job successful. {Description}", GetType().Name, job.Description());
+            RefreshWbGeometry refreshAction = new(witsmlClient.GetServerHostname(), job.Target.WellUid, job.Target.WellboreUid, job.Target.Uid, RefreshType.Update);
+            WorkerResult workerResult = new(witsmlClient.GetServerHostname(), true, $"WbGeometrySections {wbGeometrySectionsString} copied to: {targetWbGeometry.Name}");
+
+            return (workerResult, refreshAction);
+        }
+
+        private async Task<Tuple<WitsmlWbGeometry, IEnumerable<WitsmlWbGeometrySection>>> FetchData(CopyWbGeometrySectionsJob job)
+        {
+            Task<WitsmlWbGeometry> targetWbGeometryQuery = GetWbGeometry(job.Target);
+            Task<IEnumerable<WitsmlWbGeometrySection>> sourceWbGeometrySectionsQuery = GetWbGeometrySections(job.Source.Parent, job.Source.ComponentUids);
+            await Task.WhenAll(targetWbGeometryQuery, sourceWbGeometrySectionsQuery);
+            WitsmlWbGeometry targetWbGeometry = targetWbGeometryQuery.Result;
+            IEnumerable<WitsmlWbGeometrySection> sourceWbGeometrySections = sourceWbGeometrySectionsQuery.Result;
+            return Tuple.Create(targetWbGeometry, sourceWbGeometrySections);
+        }
+
+        private async Task<WitsmlWbGeometry> GetWbGeometry(ObjectReference wbGeometryReference)
+        {
+            WitsmlWbGeometrys witsmlWbGeometry = WbGeometryQueries.GetWitsmlWbGeometryById(wbGeometryReference.WellUid, wbGeometryReference.WellboreUid, wbGeometryReference.Uid);
+            WitsmlWbGeometrys result = await GetTargetWitsmlClientOrThrow().GetFromStoreAsync(witsmlWbGeometry, new OptionsIn(ReturnElements.All));
```

Comment:
This method is called twice on copy it seems one for target and one for source, so it will fail on fetching source because it uses `TargetWitsmlClient`